tests/libjlm/backend/hls/rvsdg2rhls/test-empty-gamma.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022 Nico Reißmann <nico.reissmann@gmail.com>
+ *                Magnus Sjalander <work@sjalander.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#include "test-registry.hpp"
+#include "test-types.hpp"
+
+#include <jive/rvsdg/gamma.hpp>
+#include <jive/view.hpp>
+
+#include <jlm/backend/hls/rvsdg2rhls/rvsdg2rhls.hpp>
+#include <jlm/ir/operators.hpp>
+#include <jlm/ir/hls/hls.hpp>
+#include <jlm/ir/RvsdgModule.hpp>
+
+static void
+test_with_match()
+{
+	using namespace jlm;
+
+	jlm::valuetype vt;
+	jive::bittype bt1(1);
+	FunctionType ft({&bt1, &vt, &vt}, {&vt});
+
+	RvsdgModule rm(filepath(""), "", "");
+	auto nf = rm.Rvsdg().node_normal_form(typeid(jive::operation));
+	nf->set_mutable(false);
+
+	/* Setup graph */
+
+	auto lambda = lambda::node::create(rm.Rvsdg().root(), ft, "f", linkage::external_linkage);
+
+	auto match = jive::match(1, {{0, 0}}, 1, 2, lambda->fctargument(0));
+	auto gamma = jive::gamma_node::create(match, 2);
+	auto ev1 = gamma->add_entryvar(lambda->fctargument(1));
+	auto ev2 = gamma->add_entryvar(lambda->fctargument(2));
+	auto ex = gamma->add_exitvar({ev1->argument(0), ev2->argument(1)});
+
+	auto f = lambda->finalize({ex});
+	rm.Rvsdg().add_export(f, {f->type(), ""});
+
+	jive::view(rm.Rvsdg(), stdout);
+
+	/* Convert graph to RHLS */
+
+	hls::rvsdg2rhls(rm);
+	jive::view(rm.Rvsdg(), stdout);
+
+	/* Verify output */
+
+	bool hls_mux = false;
+	// The lambda node has been replaced so need to start from
+	// the root of the graph
+	auto root = rm.Rvsdg().root();
+	auto newLambda = dynamic_cast<const jlm::lambda::node *>(root->nodes.begin().ptr());
+	if (root->nnodes() == 1 && newLambda) {
+		for (const auto & node : newLambda->subregion()->nodes) {
+			if (jive::is<jlm::hls::mux_op>(&node))
+				hls_mux = true;
+		}
+	}
+	assert(hls_mux);
+}
+
+static void
+test_without_match()
```

Comment:
test_gamma3 is also without a match as well. Could we maybe name the functions a little bit better?

Reply:
This is from one of your own test cases so what do you want to call it?

Reply:
Looking at the gamma tests. What do you test in the test_with_match unit test that you do not test in the test_gamma3 unit test? The only difference I can see is that the test_gamma3 has a gamma with 3 cases instead of 2.